rust/images/src/change_set.rs

```diff
@@ -154,7 +158,7 @@ impl ChangeSet {
             };
 
             let mut archive = tar::Builder::new(&mut layer_writer);
-
+            archive.mode(HeaderMode::Deterministic);
```

Comment:
Hi @jm-observer, I don't yet understand the issue (I'll try to add some tests to try to understand it). But does this fix the issue? As far as I can tell changing the header mode just changes the meta data https://docs.rs/tar/latest/tar/enum.HeaderMode.html

Reply:
Actually, @jm-observer would you mind providing some shell commands, or ideally some Rust tests that show the issue?

Reply:
for example, a Dockerfile is as follows. It should generate the same layer digest whenever it is built.  But not now, because the metadata of the config folder is different at different build times, resulting in different digest generated 
```
FROM scratch
COPY ./helloworld /config/helloworld
CMD /config/helloworld
```

Reply:
ok, wait for me to write a test. It's a little bit clearer

Reply:
hi, @nokome , i write a simple test for this: _https://github.com/jm-observer/images_test.git_

Reply:
Thanks you for those test @jm-observer. I have added similar test to the `change_set` module https://github.com/stencila/stencila/commit/879b3aa681a8ecd0d26f17e82f0e65d9d02e0506 and merged the PR. Thanks!